Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,543 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// A protocol for message content delegate responsible for action handling.
+///
+/// When custom message content view is created, the protocol that inherits from this one
+/// should be created if an action can be taken on the new content view.
+public protocol MessageContentViewDelegate: AnyObject {
+    /// Gets called when error indicator is tapped.
+    /// - Parameter indexPath: The index path of the cell displaying the content view. Equals to `nil` when
+    /// the content view is displayed outside the collection/table view.
+    func messageContentViewDidTapOnErrorIndicator(_ indexPath: IndexPath?)
+
+    /// Gets called when thread reply button is tapped.
+    /// - Parameter indexPath: The index path of the cell displaying the content view. Equals to `nil` when
+    /// the content view is displayed outside the collection/table view.
+    func messageContentViewDidTapOnThread(_ indexPath: IndexPath?)
+
+    /// Gets called when quoted message view is tapped.
+    /// - Parameter indexPath: The index path of the cell displaying the content view. Equals to `nil` when
+    /// the content view is displayed outside the collection/table view.
+    func messageContentViewDidTapOnQuotedMessage(_ indexPath: IndexPath?)
+}
```

Comment:
Following up on you comments about the index path, should we use `messageId` here, instead?